### DISCUSSION TITLE
Help taking images using the Python version of Cuvis SDK (Sample 5)

### DISCUSSION THREAD

**Discussion by troykerim** [cubert-hyperspectral/cuvis.sdk#56] [Q&A] [answered]
Hello,

I am a graduate student and I work on a project with a team of students that uses your Hyperspectral imaging camera.  You have corresponded with my classmates on the Jira ticketing system.  We were told that we could ask questions here on github. 

We are using the Python version of the Cuvis SDK.  We specifically have questions regarding the python sample code specifically Sample 5 (EX05_recordSingleImage).  

We have the following set up:
Operating System: Windows 10
Camera Model: Cubert S5
Language: Python
Python Virtual Environment: Yes
Is the Camera Software Setup: Yes

Note: Our virtual environment is set up based on the information provided in a repository that you had on your github repository before.  It mentioned making a virtual environment for python with the instruction pip install . and adding the setuptools package and other packages were included as well.  However, we noticed that this specific repo is no longer available. 

Our issue with Sample 5 is that there is an entire section of the code that does not execute.  Please see the attached screenshots.  We were told in our Jira conversations that this specific file would allow the camera to do the following: Calibrate itself, take a screenshot(s) and then send those screenshots to a specific file/location.  As seen in our screenshots, even if we specify a specific file path when running the code , no image or any file for that matter appears in the folder of our choice.  We have the camera connected to our computer and we checked to make sure that the camera is able to receive information by pinging it.   

When we ran the same code in C#, those lines in the code would execute and we would see it in the terminal.  We would see that print statement Camera is loading… and it would eventually connect to our camera and execute the remaining portion of the code. But for the python version it never does that.  And specifically for our needs, we need python to work. That is our language for our project. 

How can we resolve this issue?  Is there something we are missing?  We really want to take images using the SDK directly.  What do you suggest we do?  
![sample5_img2](https://github.com/cubert-hyperspectral/cuvis.sdk/assets/31581632/35563ca4-430a-4e01-a74b-b461f96a5bb6)
![sample5_img1](https://github.com/cubert-hyperspectral/cuvis.sdk/assets/31581632/e9bb2fab-dcc5-4dce-8acd-455dfeeda9e3)

**Comment by antmedellin** [comment, marked as answer]
Howdy, 

I am also a graduate student working with the Ultris S5. I work more with the c++ version of the sdk on Ubuntu, so networking is a bit different, but in the past I had a similar issue. I fixed it by making sure that the ip address of the ethernet connection is 192.168.200.x and has a subnet of 255.255.255.0. Sometimes if you use a different ethernet dongle, this can change. You can double check that their software can communicate with the camera via CuvisTouch. 

Also to get a bit more information, did you install from source or pip? If you installed from source, which commit did you use? Cuvis.sdk points to a different commit than the current main branch.

**Reply by troykerim** [reply]
Hi, 

Thank you for responding.

We are able to ping the Camera using the CuvisTouch.  The camera connection to our computer when plugged in appears to be fine.  My groupmates think that we may have to just do a fresh install of the entire SDK.  

To answer the question regarding the repo version, we originally installed it back in June and there have been several updates to this repo since then.  The old repo that we used may not even exist anymore.  For example, the instructions they provided that we followed to set up the virtual environment don’t exist in this repo anymore.  

Regarding Sample 5:
Have you used it and got it to work in its entirety?
Is it able to take images or not?
If it does take images, can it send them to a specific file location?

**Reply by antmedellin** [reply]
I did get Sample 5 working, but for cpp on Ubuntu. I am able to take images and set the name and export directory location. 
Here is a link to my repo [Hypertools](https://github.com/tamu-edu-students/HyperTools/tree/main) that has my process for doing it. Documentation is not the best, so if you need help, just reach out to me.

**Comment by robertCubert** [comment]
>  To answer the question regarding the repo version, we originally installed it back in June and there have been several updates to this repo since then. The old repo that we used may not even exist anymore. For example, the instructions they provided that we followed to set up the virtual environment don’t exist in this repo anymore.


indeed there have been quite some changes, so in order to assist you further, please completely uninstall cuvis and set up a new repo according to this: https://github.com/cubert-hyperspectral/cuvis.python 

if the problem persists, let us know!
